ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -161,7 +165,26 @@ private short getDefaultClusterReplication() {
   @Override
   public boolean isTopicExists(final String topic) {
     LOG.trace("Checking for existence of topic '{}'", topic);
-    return listTopicNames().contains(topic);
+    try {
+      ExecutorUtil.executeWithRetries(
+          () -> adminClient.get().describeTopics(
+              ImmutableList.of(topic),
+              new DescribeTopicsOptions().includeAuthorizedOperations(true)
+          ).values().get(topic).get(),
+          RetryBehaviour.ON_RETRYABLE
+      );
+      return true;
+    } catch (final Exception e) {
+      if (Throwables.getRootCause(e) instanceof UnknownTopicOrPartitionException) {
+        return false;
+      }
+
+      if (Throwables.getRootCause(e) instanceof TopicAuthorizationException) {
```

Comment:
If they're not authorized, it returns false.  Was this the behavior of listTopicNames that it returned a list of only the topics for which they are authorized?

Reply:
that's what i assumed, but I've removed this because after local testing it looks like it will actually throw a `UnknownTopicException` if you don't have view access so I'll leave it to propegate the error in this case